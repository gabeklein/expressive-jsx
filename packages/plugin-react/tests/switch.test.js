const test = require("./_adapter");

test("basic switch", `
  ({ hello }) => do {
    if(hello)
      <div>Hello World</div>
    else
      <div>Goodbye World</div>
  }
`);

test("switch styles", `
  ({ foo }) => do {
    if(foo)
      color: red;
    else
      color: blue;
  }
`);

test("alternate consequent types", `
  ({ foo }) => do {
    if(foo)
      color: red;
    else
      <div>Foo is false</div>
  }
`);

test("inverse consequent types", `
  ({ foo }) => do {
    if(foo)
      <div>Foo is true</div>
    else
      color: red;
  }
`);

test("supports complex else-if", `
  ({ foo, bar }) => do {
    if(foo)
      <div>Foo is true</div>
    else if(bar)
      <div>Bar is true</div>
    else
      color: red;
  }
`);

test("supports inverse else-if", `
  ({ foo, bar }) => do {
    if(foo)
      color: red;
    else if(bar)
      color: blue;
    else
      <div>Foo & Bar are false</div>
  }
`);




