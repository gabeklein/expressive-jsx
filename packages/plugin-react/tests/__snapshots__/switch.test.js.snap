// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alternate consequent types 1`] = `
"({
  foo
}) => create(\\"div\\", {
  \\"className\\": !!foo && \\"if_foo_1u9\\"
}, !foo && create(\\"div\\", {}, \\"Foo is false\\"));

Styles.include(\`
	.do_erd.if_foo_1u9 { color: red }
\`);"
`;

exports[`alternate consequent types 2`] = `
"({
  foo
}) => <div className={!!foo && \\"if_foo_1u9\\"}>{!foo && <div>Foo is false</div>}</div>;

Styles.include(\`
	.do_erd.if_foo_1u9 { color: red }
\`);"
`;

exports[`basic switch 1`] = `
"({
  hello
}) => hello ? create(\\"div\\", {}, \\"Hello World\\") : create(\\"div\\", {}, \\"Goodbye World\\");"
`;

exports[`basic switch 2`] = `
"({
  hello
}) => hello ? <div>Hello World</div> : <div>Goodbye World</div>;"
`;

exports[`delegated variant may still contain style 1`] = `
"const BuyButton = ({
  active
}) => create(\\"div\\", {
  \\"className\\": active ? \\"if_active_392 exampleActive_fbb\\" : \\"else_122\\"
}, \\"Hello World!\\");

Styles.include(\`
	.example_6a2.else_122 { color: green }
	.example_6a2.if_active_392 { background: white }
	.exampleActive_fbb { color: blue }
\`);"
`;

exports[`delegated variant may still contain style 2`] = `
"const BuyButton = ({
  active
}) => <div className={active ? \\"if_active_392 exampleActive_fbb\\" : \\"else_122\\"}>Hello World!</div>;

Styles.include(\`
	.example_6a2.else_122 { color: green }
	.example_6a2.if_active_392 { background: white }
	.exampleActive_fbb { color: blue }
\`);"
`;

exports[`delegates definition via use 1`] = `
"const BuyButton = ({
  active
}) => create(\\"div\\", {
  \\"className\\": join(\\"example_14k\\", active ? \\"exampleActive_fbb\\" : \\"else_122\\")
}, \\"Hello World!\\");

Styles.include(\`
	.example_14k { color: black }
	.example_6a2.else_122 { color: green }
	.exampleActive_fbb { color: blue }
\`);"
`;

exports[`delegates definition via use 2`] = `
"const BuyButton = ({
  active
}) => <div className={join(\\"example_14k\\", active ? \\"exampleActive_fbb\\" : \\"else_122\\")}>Hello World!</div>;

Styles.include(\`
	.example_14k { color: black }
	.example_6a2.else_122 { color: green }
	.exampleActive_fbb { color: blue }
\`);"
`;

exports[`inverse consequent types 1`] = `
"({
  foo
}) => create(\\"div\\", {
  \\"className\\": !foo && \\"else_7qc\\"
}, !!foo && create(\\"div\\", {}, \\"Foo is true\\"));

Styles.include(\`
	.do_erd.else_7qc { color: red }
\`);"
`;

exports[`inverse consequent types 2`] = `
"({
  foo
}) => <div className={!foo && \\"else_7qc\\"}>{!!foo && <div>Foo is true</div>}</div>;

Styles.include(\`
	.do_erd.else_7qc { color: red }
\`);"
`;

exports[`nested elements may have conditional 1`] = `
"const BuyButton = ({
  active
}) => create(\\"div\\", {}, create(\\"div\\", {}, create(\\"div\\", {
  \\"className\\": active ? \\"if_active_27p\\" : \\"else_1kq\\"
})));

Styles.include(\`
	.inner_1mn.else_1kq { color: green }
	.inner_1mn.if_active_27p { color: red }
\`);"
`;

exports[`nested elements may have conditional 2`] = `
"const BuyButton = ({
  active
}) => <div><div><div className={active ? \\"if_active_27p\\" : \\"else_1kq\\"} /></div></div>;

Styles.include(\`
	.inner_1mn.else_1kq { color: green }
	.inner_1mn.if_active_27p { color: red }
\`);"
`;

exports[`supports complex else-if 1`] = `
"({
  foo,
  bar
}) => create(\\"div\\", {
  \\"className\\": !foo && !bar && \\"else_40u\\"
}, foo ? create(\\"div\\", {}, \\"Foo is true\\") : !!bar && create(\\"div\\", {}, \\"Bar is true\\"));

Styles.include(\`
	.do_erd.else_40u { color: red }
\`);"
`;

exports[`supports complex else-if 2`] = `
"({
  foo,
  bar
}) => <div className={!foo && !bar && \\"else_40u\\"}>{foo ? <div>Foo is true</div> : !!bar && <div>Bar is true</div>}</div>;

Styles.include(\`
	.do_erd.else_40u { color: red }
\`);"
`;

exports[`supports inverse else-if 1`] = `
"({
  foo,
  bar
}) => create(\\"div\\", {
  \\"className\\": foo ? \\"if_foo_1u9\\" : !!bar && \\"if_bar_7qc\\"
}, !foo && !bar && create(\\"div\\", {}, \\"Foo & Bar are false\\"));

Styles.include(\`
	.do_erd.if_bar_7qc { color: blue }
	.do_erd.if_foo_1u9 { color: red }
\`);"
`;

exports[`supports inverse else-if 2`] = `
"({
  foo,
  bar
}) => <div className={foo ? \\"if_foo_1u9\\" : !!bar && \\"if_bar_7qc\\"}>{!foo && !bar && <div>Foo & Bar are false</div>}</div>;

Styles.include(\`
	.do_erd.if_bar_7qc { color: blue }
	.do_erd.if_foo_1u9 { color: red }
\`);"
`;

exports[`switch styles 1`] = `
"({
  foo
}) => create(\\"div\\", {
  \\"className\\": foo ? \\"if_foo_1u9\\" : \\"else_7qc\\"
});

Styles.include(\`
	.do_erd.else_7qc { color: blue }
	.do_erd.if_foo_1u9 { color: red }
\`);"
`;

exports[`switch styles 2`] = `
"({
  foo
}) => <div className={foo ? \\"if_foo_1u9\\" : \\"else_7qc\\"} />;

Styles.include(\`
	.do_erd.else_7qc { color: blue }
	.do_erd.if_foo_1u9 { color: red }
\`);"
`;
